### PR TITLE
Feature: command output

### PR DESCRIPTION
**Pull request by vadymmarkov**
@zenangst @RamonGilabert @onmyway133 

1) `Projection` is removed, so we have less confusion with new terms 😄 Now you execute command with a specified output result type, and react to event produced by this command. The flow looks like this for example:

``` swift
struct UsersStory {

  struct Command: Aftermath.Command {
    typealias Output = [ViewModel]
  }

  struct Handler: Aftermath.CommandHandler {

    func handle(command: Command) throws -> Event<Command> {
      let request = UsersRequest()

      Malibu.networking("base").GET(request)
        .validate()
        .toJSONArray()
        .then({ array -> [User] in try array.map({ try User($0) }) })
        .then({ users -> [ViewModel] in
          return users.map({ user in
            ViewModel(
              identifier: user.id,
              title: user.name.capitalizedString,
              subtitle: "Email: \(user.email)",
              action: "users:\(user.id)")
          })
        })
        .done({ items in
          self.fulfill(items)
        })
        .fail({ error in
          self.reject(error)
        })

      return progress
    }
  }
}
```

2) Reaction's disposal tokens are handled automatically, the only thing that you need to do is to call `disposeAll()` from `ReactionProducer`.

3) There is a basic implementation of an example project that uses `Spots` and a lot of other cool stuff.

### CODE REVIEW

**Comment by vadymmarkov** [comment]
`Demo` is broken now because `AftermathCompass` needs to be updated to changes in this PR 😄

**Comment by zenangst** [comment]
I like this change!

tl;dr

379a6740-2ce5-11e6-8d7f-c2ea5257f2f9
